sklearn/feature_selection/tests/test_from_model.py

```diff
@@ -40,6 +42,117 @@ def test_input_estimator_unchanged():
     assert_true(transformer.estimator is est)
 
 
+def check_invalid_max_features(est, X, y):
```

Comment:
We can parametrize the test using pytest from now on:

```python
@pytest.mark.parametrize("max_features", [-1, X.shape[1] + 1, 'gobbledigook', 'all'])
def check_invalid_max_features(est, X, y, max_features):
    transformer = SelectFromModel(estimator=est,
                                      max_features=max_features,
                                      threshold=-np.inf)
    with pytest.raises(ValueError, err_msg):
        transformer.fit(X, y)
```

Reply:
Also it is weird that you have always the same error `"max_features should be >=0"`.
It is not meaningful for string. We need to make 2 if conditions for type and values.